http.go

```diff
@@ -156,6 +157,18 @@ func main() {
 	if err := showResponse(p, resp, os.Stdout); err != nil {
 		fatalf("%v", err)
 	}
+	if p.checkStatus && resp.StatusCode >= http.StatusMultipleChoices {
```

Comment:
I think I'd do:

     statusClass := resp.StatusCode / 100
     if p.checkStatus && resp.StatusClass != 2 {
          fmt.Fprintf(os.Stderr, "warning: HTTP response code %s\n", resp.Status)
          os.Exit(statusClass)
    }

Also, I wonder if the warning should be printed before the body is copied
to stdout.